client/.eslintrc.js

```diff
@@ -18,6 +18,7 @@ module.exports = {
     'no-param-reassign': 0,
     'no-mixed-operators': 0,
     'no-underscore-dangle': 0,
+    "no-use-before-define": ["error", "nofunc"],
```

Comment:
Function hoisting can be trusted..
It's a must for useEffects to be at the top of functions.

Reply:
I think `useEffect` should be not at the top of function - it could anywhere, you just need to keep an order of hooks.

Reply:
I generally like the hooked components to have this structure:
```
import X from 'x';

const VAR1 = ...

function utilityFunc1() {
}

function utilityFunc2() {
}

function Component() {
  const [state1, setState1] = userState();
  const [state2, setState2] = userState();

  const var1 = ...
  const var2 = ...

  useEffect(() => { // init
  }, []);

  useEffect(() => {
  }, [state1]);
  
  function method1() {
  }

  function method2() {
  }

  function renderPart1() {
  }

  function renderPart2() {
  }

  return (
    {renderPart1()}
    {renderPart2()}
    ...
  );
}

Reply:
In this case I'd prefer to take as much as possible of that `method1`, `method2`, `renderPart1` function outside of the component (adding more arguments if needed).

Reply:
BTW - with class components that functions were class methods, and probably entire component structure were cleaner (IMHO :slightly_smiling_face:)

Reply:
> In this case I'd prefer to take as much as possible of that `method1`, `method2`

Yeah I keep in only what uses state in a way that would take too many arguments if it was out.

> component structure were cleaner

It's also my opinion but I wanna try to get the hang of it and keep discussing and perfecting.